mkdocs/docs/index.md

```diff
@@ -1,22 +1,156 @@
+- [x] Support for dynamic shapes, flow control, streaming, and other advanced
+model features
+- [x] Ahead-of-time compilation of scheduling and execution logic together
+- [x] Low overhead, pipelined execution for efficient power and resource usage
+- [x] Binary size as low as 30KB on embedded systems
+- [x] Debugging and profiling support
+- [ ] Support for running quantized models (coming soon!)
 
-### Frontends
+### Support matrix
 
-<!-- TODO(??): explain what frontends are, introduce options -->
+The IREE compiler tools run on :fontawesome-brands-linux: Linux,
+:fontawesome-brands-windows: Windows, and :fontawesome-brands-apple: macOS.
+IREE supports importing from a variety of ML frameworks and running models on a
+wide range of platforms, CPU architectures, and hardware accelerators.
 
-### Backends
+Model formats
 
-<!-- TODO(??): explain what backends are, introduce options -->
+- [x] TensorFlow
+- [x] TensorFlow Lite
+- [x] JAX
+- [ ] PyTorch
+<!-- - [ ] ONNX -->
 
-### Bindings
+Runtime platforms and architectures
 
-<!-- TODO(??): explain what bindings are, introduce options -->
+- [x] Linux
+- [x] Windows
+- [x] macOS, iOS
+- [x] Android
+- [x] Bare metal, including RISC-V
+- [x] ARM and x86
+- [ ] WebAssembly
+
+Accelerators and APIs
+
+- [x] Vulkan
+- [x] CUDA
+- [ ] Metal
+- [ ] WebGPU
+
+### Project architecture
+
+IREE adopts a _holistic_ approach towards ML model compilation: the IR produced
+contains both the _scheduling_ logic, required to communicate data dependencies
+to low-level parallel pipelined hardware/API like
+[Vulkan](https://www.khronos.org/vulkan/), and the _execution_ logic, encoding
+dense computation on the hardware in the form of hardware/API-specific binaries
+like [SPIR-V](https://www.khronos.org/spir/).
+
+<!-- TODO(scotttodd): text borders so this is easier to read on dark backgrounds -->
+<!-- TODO(scotttodd): use the same .svg as the README (point README at this path) -->
+![IREE Architecture](assets/images/iree_architecture.svg)
+
+## Workflow overview
+
+Using IREE involves these general steps:
+
+1. **Import your model**
+
+    Work in your framework of choice, then run your model through one of IREE's
+    import tools.
+
+2. **Select your deployment configuration**
+
+    Identify your target platform, accelerator(s), and other constraints.
+
+3. **Compile your model**
+
+    Compile through IREE, picking compilation targets based on your
+    deployment configuration.
+
+4. **Run your model**
+
+    Use IREE's runtime components to execute your compiled model.
+
+### Importing models from ML frameworks
+
+IREE supports importing models from multiple ML frameworks and model formats.
+
+<!-- TODO(#5555): rename "frontends" to "ML frameworks"? -->
+See the "Frontends" pages for guidance on working with each framework and
+model format.
+
+### Selecting deployment configurations
+
+IREE provides a flexible set of tools for various deployment scenarios - use as
+much or as little of IREE as you like.
```

Comment:
s/use as much or as little of IREE as you like/only pay the cost for use/ to be more positive. :)

Reply:
Thought about that, but didn't want to use "pay for what you use" since people might think we're monetizing / charging for features :P

Will think more on phrasing...

Reply:
Reworded:

> IREE provides a flexible set of tools for various deployment scenarios.
Fully featured environments can use IREE for dynamic model deployments taking
advantage of multi-threaded hardware, while embedded systems can bypass IREE's
runtime entirely or interface with custom accelerators.